### ISSUE TITLE
Can you explain the preprocess?

### ISSUE THREAD

**Issue by zhangjh915** [mozilla/LPCNet#82]
I found there are many tricks in data preprocessing. For example, the smoothing at the silence boundary, the random noise added. Can you summarize the pipeline of the preprocess and explain the tricks that you use? Are all those tricks useful for model training? Thank you!

**Comment by chengshaodi** [comment]
@zhangjh915 How to get the  data preprocessing? i canot find the code.thanks

**Comment by zhangjh915** [comment]
> @zhangjh915 How to get the data preprocessing? i canot find the code.thanks

You can use "dump_data" directly following README with input wav.s16
And the code is in "src/dump_data.c"

**Comment by chengshaodi** [comment]
bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mozilla/LPCNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月18日(星期六) 上午10:42
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82)





  
@zhangjh915 How to get the data preprocessing? i canot find the code.thanks
  
You can use "dump_data" directly following README with input wav.s16
 And the code is in "src/dump_data.c"
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zhangjh915** [comment]
> bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗?
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" ***@***.***&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same).

**Comment by chengshaodi** [comment]
Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't.
Can you tell  me  this approach is feasible?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mozilla/LPCNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月1日(星期一) 中午11:15
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82)





  
bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗?
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
  
OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same).
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zhangjh915** [comment]
> Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't. Can you tell me this approach is feasible?
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" ***@***.***&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 中午11:15 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same). — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

I simply used np.fromfile(feature_file, dtype='float32') to read the feature file.

**Comment by chengshaodi** [comment]
Me too,then i used np.reshape(55,),why&nbsp;Why can't it be divided into 55d? it's my features is error? The 55d lenth is same lenth?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mozilla/LPCNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月1日(星期一) 下午2:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82)





  
Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't. Can you tell me this approach is feasible?
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 中午11:15 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same). — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
  
I simply used np.fromfile(feature_file, dtype='float32') to read the feature file.
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zhangjh915** [comment]
> Me too,then i used np.reshape(55,),why&nbsp;Why can't it be divided into 55d? it's my features is error? The 55d lenth is same lenth?
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" ***@***.***&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:19 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't. Can you tell me this approach is feasible? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 中午11:15 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same). — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. I simply used np.fromfile(feature_file, dtype='float32') to read the feature file. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

You mean the length cannot be divided by 55? Then just get rid of the rest of the values. Check the older version I used. https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L76

**Comment by chengshaodi** [comment]
When i used './dump_data -train input.s16 features.f32 data.s16'&nbsp; to get the features.f32,the file lenth is 180000000,so i coudn't to reshape 55d.I just want to remove 18d(BFE).What should I do?Thanks a lot.






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mozilla/LPCNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月1日(星期一) 下午2:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82)





  
Me too,then i used np.reshape(55,),why&nbsp;Why can't it be divided into 55d? it's my features is error? The 55d lenth is same lenth?
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:19 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't. Can you tell me this approach is feasible? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 中午11:15 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same). — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. I simply used np.fromfile(feature_file, dtype='float32') to read the feature file. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
  
You mean the length cannot be divided by 55? Then just get rid of the rest of the values. Check the older version I used. https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L76
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zhangjh915** [comment]
> When i used './dump_data -train input.s16 features.f32 data.s16'&nbsp; to get the features.f32,the file lenth is 180000000,so i coudn't to reshape 55d.I just want to remove 18d(BFE).What should I do?Thanks a lot.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" ***@***.***&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:48 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Me too,then i used np.reshape(55,),why&nbsp;Why can't it be divided into 55d? it's my features is error? The 55d lenth is same lenth? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:19 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't. Can you tell me this approach is feasible? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 中午11:15 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same). — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. I simply used np.fromfile(feature_file, dtype='float32') to read the feature file. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. You mean the length cannot be divided by 55? Then just get rid of the rest of the values. Check the older version I used. https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L76 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

Can you read and try the method in this code? https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L59-L78

**Comment by chengshaodi** [comment]
Of course,but the features.f32&nbsp; file comes from './dump_data -train input.s16 features.f32 data.s16',then I write to code 'np.fromfile('.f32',np.float32), np.shape(55,)',the code is error.
&nbsp;Why&nbsp; should see train_lpcnet?&nbsp;






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mozilla/LPCNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月1日(星期一) 下午3:10
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82)





  
When i used './dump_data -train input.s16 features.f32 data.s16'&nbsp; to get the features.f32,the file lenth is 180000000,so i coudn't to reshape 55d.I just want to remove 18d(BFE).What should I do?Thanks a lot.
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:48 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Me too,then i used np.reshape(55,),why&nbsp;Why can't it be divided into 55d? it's my features is error? The 55d lenth is same lenth? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:19 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't. Can you tell me this approach is feasible? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 中午11:15 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same). — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. I simply used np.fromfile(feature_file, dtype='float32') to read the feature file. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. You mean the length cannot be divided by 55? Then just get rid of the rest of the values. Check the older version I used. https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L76 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
  
Can you read and try the method in this code? https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L59-L78
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zhangjh915** [comment]
> Of course,but the features.f32&nbsp; file comes from './dump_data -train input.s16 features.f32 data.s16',then I write to code 'np.fromfile('.f32',np.float32), np.shape(55,)',the code is error. &nbsp;Why&nbsp; should see train_lpcnet?&nbsp;
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" ***@***.***&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午3:10 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) When i used './dump_data -train input.s16 features.f32 data.s16'&nbsp; to get the features.f32,the file lenth is 180000000,so i coudn't to reshape 55d.I just want to remove 18d(BFE).What should I do?Thanks a lot. … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:48 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Me too,then i used np.reshape(55,),why&nbsp;Why can't it be divided into 55d? it's my features is error? The 55d lenth is same lenth? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:19 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't. Can you tell me this approach is feasible? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 中午11:15 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same). — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. I simply used np.fromfile(feature_file, dtype='float32') to read the feature file. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. You mean the length cannot be divided by 55? Then just get rid of the rest of the values. Check the older version I used. https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L76 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. Can you read and try the method in this code? https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L59-L78 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

Please check my previous replies and the code. Before reshape, there is another line "features = features[:nb_frames*feature_chunk_size*nb_features]". So it must be divided by nb_features, which is 55.

**Comment by chengshaodi** [comment]
thank you so much.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mozilla/LPCNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月1日(星期一) 下午3:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82)





  
Of course,but the features.f32&nbsp; file comes from './dump_data -train input.s16 features.f32 data.s16',then I write to code 'np.fromfile('.f32',np.float32), np.shape(55,)',the code is error. &nbsp;Why&nbsp; should see train_lpcnet?&nbsp;
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午3:10 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) When i used './dump_data -train input.s16 features.f32 data.s16'&nbsp; to get the features.f32,the file lenth is 180000000,so i coudn't to reshape 55d.I just want to remove 18d(BFE).What should I do?Thanks a lot. … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:48 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Me too,then i used np.reshape(55,),why&nbsp;Why can't it be divided into 55d? it's my features is error? The 55d lenth is same lenth? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 下午2:19 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) Thank you.I get the features file from&nbsp;./dump_data -train input.s16 features.f32 data.s16.Then i want to use numpy to reshape the file to 55d,but i can't. Can you tell me this approach is feasible? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年11月1日(星期一) 中午11:15 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) bro，I get the features.f32 file,i did not divide it into 55d？the file lenth is 180000000.我可以说中文吗? … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mozilla/LPCNet" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 上午10:42 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [mozilla/LPCNet] Can you explain the preprocess? (#82) @zhangjh915 How to get the data preprocessing? i canot find the code.thanks You can use "dump_data" directly following README with input wav.s16 And the code is in "src/dump_data.c" — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. OK, feel free to do so. The feature file needs to be reshaped to 55. And the code has done it for you. Check https://github.com/mozilla/LPCNet/blob/master/training_tf2/train_lpcnet.py (This code is different from what I used but the idea should be the same). — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. I simply used np.fromfile(feature_file, dtype='float32') to read the feature file. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. You mean the length cannot be divided by 55? Then just get rid of the rest of the values. Check the older version I used. https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L76 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. Can you read and try the method in this code? https://github.com/mozilla/LPCNet/blob/34d82a9114d97d34489d2dcf6851a925cd4a2f93/src/train_lpcnet.py#L59-L78 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
  
Please check my previous replies and the code. Before reshape, there is another line "features = features[:nb_framesfeature_chunk_sizenb_features]". So it must be divided by nb_features, which is 55.
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by RookieImage** [comment]
@chengshaodi Excuse me, how did you get the input s16 file? I have downloaded the datasets in datasets.txt but they are all wav.

**Comment by RookieImage** [comment]
I have seen the problem in other issues, thx
